debian/gg-gamesave.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" (C) Copyright 2022 Landon H. <100165458+LandonTheCoder@users.noreply.github.com>,
.\"
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH GG-GAMESAVE 1 "April 17 2022"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
gg-gamesave \- program to open Group Guess savefiles

.SH SYNOPSIS
.B gg-gamesave
.RI [ options ] " file"

.SH DESCRIPTION
This manual page documents briefly the
.B gg-gamesave
command.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.

\fBgg-gamesave\fP is a program that interprets JSON savefiles for
.IR "Group Guess" ,
and loads a game window with the gamefile's specified questions and answers.

.SH OPTIONS
These programs follow the usual GNU command line syntax, with long options
starting with two dashes (`-'). A summary of options is included below.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-verbose
Tell
.B gg-gamesave
to give more verbose output, useful for debugging. This overrides the "debug" setting in the savefile, and specifying
.B \-\-verbose
multiple times will increase verbosity by 1 (meaning more debugging output)
for each occurence of
.BR \-\-verbose .

.\" This function not implemented yet.
.\".TP
.\".B \-V, \-\-version
.\"Show version of program.

.SH SEE ALSO
The programs have documentation installed systemwide at
.IR "/usr/share/doc/group-guess" ", and examples installed at " "/usr/share/doc/group-guess/examples" ,
also available in the source code.
.PP
The "example.json" file in the systemwide examples.
.PP
The "SAVE-FORMAT.md.gz" (or "SAVE-FORMAT.md") file in the
documentation directory. 
.br
Alternatively, the latest version of the
documentation on
.UR https://github.com/LandonTheCoder/group-guess/blob/main/SAVE-FORMAT.md
the group-guess GitHub page
.UE .
.PP
.\" Link of [JSON on Wikipedia](https://en.wikipedia.org/wiki/JSON)
.UR https://en.wikipedia.org/wiki/JSON
JSON on Wikipedia
.UE .


debian/changelog
group-guess (1.3.2-1) unstable; urgency=medium
  * Fix installation of manpage in RPM spec file
  
 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Sun, 16 Apr 2023 13:08:15 -0400

group-guess (1.3.1-1) unstable; urgency=medium
  * Fixes Group Guess RPM (hopefully)
  * (Attempt to) Introduce PKGBUILD for Arch
  * Fixes alignment/padding of Question label
  * Correct name of command in gg-gamesave.1 manpage to gg-gamesave
  * Use non-Ubuntu-specific icon name for the "search" icon

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Sun, 16 Apr 2023 11:10:03 -0400

group-guess (1.3.0-1) unstable; urgency=medium
  * Releases first Group Guess RPM (for Fedora)
  * Fixes bug in asset-directory detection. Now, Group Guess tests for the existence of one of its asset files in that directory.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Tue, 05 Jul 2022 17:06:03 -0400

group-guess (1.2.3-1) unstable; urgency=medium
  * Polishes port to setuptools.
  * Should be ready for production.
  * Adds a .gitignore entry for the "Group_Guess.egg-info" directory.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Fri, 13 May 2022 18:37:05 -0400

group-guess (1.2.2-2) unstable; urgency=medium
  * Attempt to port to setuptools for future-proofing.
  * This is a local release for testing.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Fri, 13 May 2022 17:14:22 -0400

group-guess (1.2.2-1) unstable; urgency=medium
  * This adds a proper manpage, which I'll have to see if it includes
    automatically.
  * This fixes a long-standing bug where the cover images had an object
    slightly longer than SVG canvase size, causing the bottom to get cut off.
  * This fixes an underwater Unicode handling bug by changing answer
    comparison from x.lower().strip() to x.casefold().strip(), which should
    catch some non-matching sequences for comparison.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Tue, 10 May 2022 15:36:48 -0400

group-guess (1.2.1-1) unstable; urgency=medium
  * This is the public version containing the fixes of 1.2.0-2 and 1.2.0-3.
    Those fixes include:
  * Add tracking of example files in debian/group-guess.examples
  * Add tracking of documentation files README.md and SAVE-FORMAT.md in
    debian/docs
  * Support varied verbosity levels. This is implemented by a new option to
    the debug logger, and the user can specify --verbose 1 or more times to
    override an input file.
  * Clean out a few unnecessary example files in the debian/ directory.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Mon, 02 May 2022 15:18:45 -0400

group-guess (1.2.0-3) unstable; urgency=medium
  * Note documentation files, so they should install into the package properly.
  * This is a version I had locally for testing.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Sun, 01 May 2022 17:42:21 -0400

group-guess (1.2.0-2) unstable; urgency=medium

  * Attempt to support examples.
  * This is a version I had locally for testing.

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Sun, 01 May 2022 17:26:48 -0400

group-guess (1.2.0-1) unstable; urgency=medium

  * Corrects debian/ references, and makes portable instances work correctly.
  * Errata: the changelog declared initial release as 1.1.2-1, but GitHub
    declared as 1.1.3 Beta.
  
 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Mon, 18 Apr 2022 17:40:40 -0400
# Fix this with the result of `date -R` later.

group-guess (1.1.2-1) unstable; urgency=medium

  * Initial release

 -- LandonTheCoder <100165458+LandonTheCoder@users.noreply.github.com>  Sun, 17 Apr 2022 12:27:48 -0400


gg-gamesave.py
#! /usr/bin/python3
"""Very basic example code.

Meant to be used as a guide for new games, until a savefile is implemented."""

import group_guess
import gi; gi.require_version("Gtk", "3.0")
from gi.repository import Gtk as gtk
import json
from group_guess import debug

class SavedWindow(group_guess.AppWindow):
  def __init__(self, filename, debug=None):
    self.json_fname = filename
    super().__init__(debug)
  def create_questions(self):
    """Create game questions."""
    # Note about json API:
    #>>> json.loads('["foo", {"bar":["baz", null, 1.0, 2]}]')
    #['foo', {'bar': ['baz', None, 1.0, 2]}]
    f = open(self.json_fname, "r")
    debug(f)
    json_file = json.load(f)
    f.close()
    self.game_title = json_file["game_title"]
    if json_file["debug"] is not None:
      group_guess_debug = json_file["debug"]
      group_guess.group_guess_debug = json_file["debug"]
    tmp_questions = json_file["questions"]
    tmp_questions_obj = []
    debug("json_file = %s" %(json_file))
    for x in tmp_questions:
      tmp_answers = []
      for y in x["answers"]:
        debug("answer_create: y = %s" %(y))
        tmp_answers.append(group_guess.Answer(**y))
      debug("question_create: question=%s," %(x["question"]),
            "the_id = %s" %(x["the_id"]),
            "answers = %s" %(tmp_answers))
      tmp_questions_obj.append(group_guess.Question(x["question"],
                                                    x["the_id"],
                                                    tmp_answers))
    debug("tmp_questions_obj = %s" %(tmp_questions_obj))
    # Need to define self.questions
    self.questions = tmp_questions_obj
  # You can define do_subclass_make_widgets() here, although it's optional.

def example_test():
  from sys import path
  test_window = SavedWindow(path[0] + "/example.json", debug=True)
  gtk.main()
def main(passed_args=None):
  import argparse
  parser = argparse.ArgumentParser(description="Load and play a saved Group Guess game")
  parser.add_argument("file", help="The savefile to open")
  args = parser.parse_args(passed_args)
  test_window = SavedWindow(args.file, debug=True)
  gtk.main()
  
main()
